src/Atmos/Model/moisture.jl

```diff
@@ -0,0 +1,95 @@
+#### Moisture component in atmosphere model
+abstract type MoistureModel end
+
+vars_state(::MoistureModel, T) = Tuple{}
+vars_transform(::MoistureModel, T) = Tuple{}
+vars_diffusive(::MoistureModel, T) = Tuple{}
+vars_aux(::MoistureModel, T) = Tuple{}
+
+function diffusive!(::MoistureModel, diffusive, ∇transform, state, aux, t, ν)
+end
+function flux_diffusive!(::MoistureModel, flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+end
+function update_aux!(::MoistureModel, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+end
+function gradtransform!(::MoistureModel, transform::Vars, state::Vars, aux::Vars, t::Real)
+end
+
+
+
+struct EquilMoist <: MoistureModel
+end
+vars_state(::EquilMoist,T) = NamedTuple{(:ρq_tot,),Tuple{T}}
+vars_transform(::EquilMoist,T) = NamedTuple{(:q_tot),Tuple{T}}
+vars_diffusive(::EquilMoist,T) = NamedTuple{(:ρd_q_tot, :ρJ_ρD), Tuple{SVector{3,T},SVector{3,T}}}
+vars_aux(::EquilMoist,T) = NamedTuple{(:e_int, :temperature),Tuple{T,T}}
+
+
+function pressure(::EquilMoist, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  q_tot = stat.ρq_tot / ρ
+  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  air_pressure(phase)
+end
+function soundspeed(::EquilMoist, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  q_tot = stat.ρq_tot / ρ
+  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  soundspeed_air(phase)
+end
+
+function gradtransform!(::EquilMoist, transform::Vars, state::Vars, aux::Vars, t::Real)
+  ρinv = 1 / state.ρ
+  transform.moisture.q_tot = ρinv * state.moisture.ρq_tot
+  transform.moisture.temperature = aux.moisture.temperature
+end
+
+
+function diffusive!(::EquilMoist, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real, ν::Union{Real,AbstractMatrix})
+  # turbulent Prandtl number
+  diag_ν = ν isa Real ? ν : diag(ν) # either a scalar or vector
+  D_q_tot = diag_ν / Prandtl_t # either a scalar or vector
```

Comment:
It would be good to keep this a bit more general. The turbulent Prandtl number may in the end be a function. So you could just define it via a function, which for now just returns a constant value. I would also make it explicit that we are using one diffusivity for moisture and temperature. Just call it `D` or `Diffusivity`, without subscript, and use it both for moisture and temperature.